src/dynamic_boundary_conditions/sea_level_rise_data.py

```diff
@@ -1,95 +1,173 @@
-                df = pd.read_csv(file_path + filename, engine='pyarrow', dtype={'siteId': str})
-                print('LOADED CSV file with pyarrow: ', filename)
-            except (Exception, ):
-                df = pd.read_csv(file_path + filename, dtype={'siteId': str})
-                print('LOADED CSV file: ', filename)
-            df["Region"] = filename[24:-4]
-            df_list.append(df)
-    return pd.concat(df_list, ignore_index=True, axis=0)
-
-
-def get_nearest_coordinate(df: pd.DataFrame, target: tuple) -> tuple:
-    """ find the nearest coordinate to target position in the input dataframe."""
-    # get unique coordinate
-    df_site = df.copy()[['lat', 'lon']].drop_duplicates(subset=['lat', 'lon'])
-    # protect input boundary for haversine function
-    assert -90 <= target[0] <= 90, "Latitude is out of range [-90, 90]"
-    assert -180 <= target[0] <= 180, "Longitude is out of range [-180, 180]"
-    # calculate distance from target coordinate
-    df_site['distance'] = df_site.apply(lambda x: (haversine(target, (x['lat'], x['lon']))), axis=1)
-    # sort by distance
-    df_site = df_site.sort_values(by=['distance'], ascending=True).reset_index(drop=True)
-    print("Nearest coordinates: ({}, {})".format(df_site.iloc[0]['lat'], df_site.iloc[0]['lon']))
-    return df_site.iloc[0]['lat'], df_site.iloc[0]['lon']
-
-
-def gen_slr_data(df: pd.DataFrame, target: tuple, crs: str = "epsg:2193") -> gpd.GeoDataFrame:
-    """ generate geopandas dataframe that the coordinate is equal to target position."""
-    # filter the coordinates based on target coordinate.
-    df = df.loc[(df['lat'] == target[0]) & (df['lon'] == target[1])]
-    # convert to geopandas dataframe from pandas dataframe
-    gdf = gpd.GeoDataFrame(df, crs=crs, geometry=gpd.points_from_xy(df.lon, df.lat))
-    gdf.rename(columns={'siteId': 'Site_ID', 'year': 'Projected_Year',
-                        'p17': 'SLR_p17_confidence_interval_meters',
-                        'p50': 'SLR_p50_confidence_interval_meters',
-                        'p83': 'SLR_p83_confidence_interval_meters',
-                        'lon': 'Longitude', 'lat': 'Latitude',
-                        'measurementName': 'Scenario_(VLM_with_confidence)', },
-               inplace=True)
-    return gdf
-
-
-def save_to_database(connect, gdf: gpd.GeoDataFrame, table_name: str, if_exists: str = 'replace'):
-    """ save dataframe to database """
-    gdf.to_postgis(table_name, connect, index=False, if_exists=if_exists)
+import pandas as pd
+import pyarrow.csv as csv
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+formatter = logging.Formatter("%(levelname)s:%(asctime)s:%(name)s:%(message)s")
+stream_handler = logging.StreamHandler()
+stream_handler.setFormatter(formatter)
+
+log.addHandler(stream_handler)
+
+
+class InvalidDirectoryError(Exception):
+    pass
+
+
+def get_catchment_area_coords(catchment_file: pathlib.Path) -> Tuple[float, float]:
```

Comment:
I think this should be called `get_area_centroid_coords` or similar

Reply:
good idea. changed to `get_catchment_centroid_coords`